workers/job-room-forwardpaginate.go

```diff
@@ -16,28 +16,42 @@ package workers
 
 import (
 	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/matrix-static/mxclient"
+	"sort"
 	"sync"
 	"time"
 )
 
 // This Job has no Resp.
 
 type RoomForwardPaginateJob struct {
-	Wg *sync.WaitGroup
+	Wg      *sync.WaitGroup
+	TTL     time.Duration
+	KeepMin int
 }
 
-const LastAccessDiscardDuration = 30 * time.Minute
-
 func (job RoomForwardPaginateJob) Work(w *Worker) {
-	// discard old rooms first
 	numRoomsBefore := len(w.rooms)
-	for id, room := range w.rooms {
-		if room.LastAccess.Before(time.Now().Add(-LastAccessDiscardDuration)) {
-			delete(w.rooms, id)
+	rooms := make([]*mxclient.Room, 0, numRoomsBefore)
+
+	// create a slice of rooms ordered by LastAccess descending
+	for _, room := range w.rooms {
+		rooms = append(rooms, room)
+	}
+	sort.Slice(rooms, func(i, j int) bool {
+		return rooms[i].LastAccess.After(rooms[j].LastAccess)
+	})
```

Comment:
Seems like this work should also be skipped if `job.KeepMin > numRoomsBefore`?

Reply:
good point